test/module-hooks/test-module-hooks-builtin-require.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const common = require('../common');
+const { registerHooks } = require('module');
+
+const schemelessBlockList = new Set([
+  'sea',
+  'sqlite',
+  'test',
+  'test/reporters',
+]);
+
+const testModules = [];
+for (const mod of schemelessBlockList) {
+  testModules.push(`node:${mod}`);
+}
+
+const hook = registerHooks({
+  resolve: common.mustCall((specifier, context, nextResolve) => nextResolve(specifier, context), testModules.length),
+  load: common.mustCall((url, context, nextLoad) => nextLoad(url, context), testModules.length),
```

Comment:
```suggestion
const hook = registerHooks({
  load: common.mustCall(function load(url, context, nextLoad) {
    assert.match(url, /^node:/);
    assert.strictEqual(schemelessBlockList.has(url.slice(5, url.length)), true);
    const result = nextLoad(url, context);
    assert.strictEqual(result.source, null);
    return {
      source: 'throw new Error("I should not be thrown because the loader ignores user-supplied source for builtins")',
      format: 'builtin',
    };
  }, testModules.length),
});
```